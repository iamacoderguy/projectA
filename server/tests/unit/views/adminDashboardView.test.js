
const each = require('jest-each').default;
const pug = require('pug');
const renderPug = (view, data) => pug.renderFile(view, data);
const Client = require('../../../models/client');

describe('adminDashboardView', () => {
    const client1 = {};
    client1.getIpAddr = () => '111.222.333.444';
    client1.getStatus = () => 'status1';

    const client2 = {};
    client2.getIpAddr = () => '111.222.333.555';
    client2.getStatus = () => 'status2';

    const client3 = {};
    client3.getIpAddr = () => '111.222.333.666';
    client3.getStatus = () => 'status3';

    each([
        [[client1, client2, client3]],
        [[]]
    ]).it('should render adminDashboardView correctly in case %o', (clients) => {
        expect(renderPug('views/adminDashboardView.pug', {
            path: 'path-to-files',
            dashboardName: 'Dashboard',
            ipAddress: '192.168.1.1',
            port: 5000,
            clients: clients,
            pin: 123456
          })).toMatchSnapshot();
    })
})