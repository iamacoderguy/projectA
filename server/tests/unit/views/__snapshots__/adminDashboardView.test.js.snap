// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adminDashboardView should render adminDashboardView correctly in case [ [length]: 0 ] 1`] = `"<!DOCTYPE html><html lang=\\"en-US\\"><head><title>Dashboard</title><meta charset=\\"utf-8\\"><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\"><meta name=\\"description\\" content=\\"Dashboard\\"><link rel=\\"stylesheet\\" href=\\"w3.css\\"></head><body><div class=\\"w3-container\\"><div class=\\"w3-container w3-dark-teal\\"><h2>Dashboard - 192.168.1.1:5000</h2></div><p><iframe width=\\"100%\\" height=\\"200vw\\" title=\\"Result box\\" srcdoc=\\"&lt;i&gt;Result...&lt;/i&gt;\\" name=\\"iframe_result\\"></iframe></p><form class=\\"w3-container\\" method=\\"POST\\" action=\\"/api/files/path?_method=PUT\\" target=\\"iframe_result\\"><p><label class=\\"w3-text-dark-teal\\" for=\\"sharedPathInput\\"><b>Set a new shared path (/api/files/path)</b></label><input class=\\"w3-input w3-border w3-round\\" id=\\"sharedPathInput\\" name=\\"path\\" type=\\"text\\" placeholder=\\"path-to-files\\" value=\\"path-to-files\\"></p><p><button class=\\"w3-btn w3-dark-teal\\" type=\\"submit\\">PUT</button></p></form><ul class=\\"w3-ul\\"><li><h2>List of clients:</h2></li><li class=\\"w3-bar\\"><p class=\\"w3-bar-item\\">There is no client</p></li></ul></div></body></html>"`;

exports[`adminDashboardView should render adminDashboardView correctly in case [ { getIpAddr: { [Function] [length]: 0, [name]: '' },
    getStatus: { [Function] [length]: 0, [name]: '' } },
  { getIpAddr: { [Function] [length]: 0, [name]: '' },
    getStatus: { [Function] [length]: 0, [name]: '' } },
  { getIpAddr: { [Function] [length]: 0, [name]: '' },
    getStatus: { [Function] [length]: 0, [name]: '' } },
  [length]: 3 ] 1`] = `"<!DOCTYPE html><html lang=\\"en-US\\"><head><title>Dashboard</title><meta charset=\\"utf-8\\"><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\"><meta name=\\"description\\" content=\\"Dashboard\\"><link rel=\\"stylesheet\\" href=\\"w3.css\\"></head><body><div class=\\"w3-container\\"><div class=\\"w3-container w3-dark-teal\\"><h2>Dashboard - 192.168.1.1:5000</h2></div><p><iframe width=\\"100%\\" height=\\"200vw\\" title=\\"Result box\\" srcdoc=\\"&lt;i&gt;Result...&lt;/i&gt;\\" name=\\"iframe_result\\"></iframe></p><form class=\\"w3-container\\" method=\\"POST\\" action=\\"/api/files/path?_method=PUT\\" target=\\"iframe_result\\"><p><label class=\\"w3-text-dark-teal\\" for=\\"sharedPathInput\\"><b>Set a new shared path (/api/files/path)</b></label><input class=\\"w3-input w3-border w3-round\\" id=\\"sharedPathInput\\" name=\\"path\\" type=\\"text\\" placeholder=\\"path-to-files\\" value=\\"path-to-files\\"></p><p><button class=\\"w3-btn w3-dark-teal\\" type=\\"submit\\">PUT</button></p></form><ul class=\\"w3-ul\\"><li><h2>List of clients:</h2></li><li class=\\"w3-bar\\"><div class=\\"w3-bar-item\\">111.222.333.444: status1</div></li><li class=\\"w3-bar\\"><div class=\\"w3-bar-item\\">111.222.333.555: status2</div></li><li class=\\"w3-bar\\"><div class=\\"w3-bar-item\\">111.222.333.666: status3</div></li></ul></div></body></html>"`;
